pallets/ocw-simple/src/lib.rs

```diff
@@ -0,0 +1,117 @@
+// Copyright (C) Moondance Labs Ltd.
+// This file is part of Tanssi.
+
+// Tanssi is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// Tanssi is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with Tanssi.  If not, see <http://www.gnu.org/licenses/>
+
+#![cfg_attr(not(feature = "std"), no_std)]
+use frame_system::pallet_prelude::BlockNumberFor;
+// Remove comment to enable http requests
+// use sp_runtime::offchain::{http, Duration};
+
+pub use pallet::*;
+#[frame_support::pallet]
+pub mod pallet {
+    use super::*;
+    use frame_support::pallet_prelude::*;
+    #[pallet::pallet]
+    pub struct Pallet<T>(_);
+
+    #[pallet::config]
+    pub trait Config: frame_system::Config {
+        /// The overarching event type.
+        type RuntimeEvent: From<Event<Self>> + IsType<<Self as frame_system::Config>::RuntimeEvent>;
+    }
+
+    #[pallet::hooks]
+    impl<T: Config> Hooks<BlockNumberFor<T>> for Pallet<T> {
+        /// Offchain worker entry point.
+        ///
+        /// By implementing `fn offchain_worker` you declare a new offchain worker.
+        /// This function will be called when the node is fully synced and a new best block is
+        /// successfully imported.
+        /// Note that it's not guaranteed for offchain workers to run on EVERY block, there might
+        /// be cases where some blocks are skipped, or for some the worker runs twice (re-orgs),
+        /// so the code should be able to handle that.
+        fn offchain_worker(_block_number: BlockNumberFor<T>) {
+            log::info!("Entering off-chain worker.");
+            // The entry point of your code called by offchain worker
+            Self::emit_offchain_event();
+            // Remove comment if you want to emit event on simple http request
+            //let _ = Self::fetch_price(block_number);
+        }
+    }
+    #[pallet::call]
+    impl<T: Config> Pallet<T> {}
+
+    /// Events for the pallet.
+    #[pallet::event]
+    #[pallet::generate_deposit(pub(super) fn deposit_event)]
+    pub enum Event<T: Config> {
+        /// Simple offchain event
+        SimpleOffchainEvent,
+        /// Event generated when new price is fetched
+        PriceFetched { block_number: BlockNumberFor<T> },
+    }
+}
+
+impl<T: Config> Pallet<T> {
+    /// Simple event emitter
+    fn emit_offchain_event() {
+        Self::deposit_event(Event::SimpleOffchainEvent);
+    }
+
+    // Remove comment to enable http requests
+    /*
+    /// Fetch current price and return the result in cents.
```

Comment:
this should be removed right?

Reply:
Yep not needed unless we want to specifically test http interactions

Reply:
Remove in https://github.com/moondance-labs/tanssi/pull/723/commits/e7ff792fad52b899b70dfb355ee15387442125bb